runner_test.go

```diff
@@ -1,48 +1,57 @@
 package main
 
 import (
-	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/http/httptest"
+	"os"
 	"testing"
 )
 
-type fakeFlagParser struct{}
-
-func (f fakeFlagParser) StringSlice(s string) []string {
-	switch s {
-	case "tags":
-		return []string{"foo", "bar"}
-	default:
-		panic(fmt.Sprintf("fakeFlagParser does not expect argument: %s", s))
-	}
+func newTestPostServer(expectedBody string, resp string, statusCode int, t *testing.T) *httptest.Server {
+	return httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		body, _ := ioutil.ReadAll(r.Body)
+		flagParams := string(body)
+		if flagParams != expectedBody {
```

Comment:
Checking the actual body string seems possibly fragile, and prevents you from checking individual fields. Instead, I would probably pass in a function to `newTestPostServer`, like `check func([]byte)`, so that the actual condition handling is handled in the outer function.